contrib/re_parser/src/re_app.nit

```diff
@@ -0,0 +1,148 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module re_app
+
+import re_parser
+import popcorn
+import popcorn::pop_config
+
+class REHandler
+	super Handler
+
+	redef fun get(req, res) do
+		res.html new RETemplate
+	end
+
+	redef fun post(req, res) do
+		# TODO Retrieve re from post body
+		var re = req.string_arg("re")
+
+		var tpl = new RETemplate
+
+		if re == null or re.is_empty then
+			tpl.error = "Error: empty regexp"
+			res.html tpl
+			return
+		end
+
+		tpl.re = re
+
+		# Parse re
+		var l = new Lexer_re_parser(re)
+		var p = new Parser_re_parser
+		p.tokens.add_all l.lex
+
+		var node = p.parse
+
+		# return error
+		if not node isa NProd then
+			if node isa NError then
+				tpl.error = "Error: {node.message}"
+			else
+				tpl.error = "Error: unable to parse regexp"
+			end
+			res.html tpl
+			return
+		end
```

Comment:
I think the whole lexer->parse->re2nfa can be put in a single method in `re_parser`, or something